xbmc/pvr/recordings/PVRRecordings.cpp

```diff
@@ -346,17 +345,18 @@ bool CPVRRecordings::GetDirectory(const std::string& strPath, CFileItemList &ite
   std::string strDirectoryPath = url.GetFileName();
   URIUtils::RemoveSlashAtEnd(strDirectoryPath);
 
-  if (StringUtils::StartsWith(strDirectoryPath, "recordings"))
+  if (StringUtils::StartsWith(strDirectoryPath, PVR_RECORDING_BASE_PATH))
   {
-    strDirectoryPath.erase(0, 10);
+    strDirectoryPath.erase(0, sizeof(PVR_RECORDING_BASE_PATH) - 1);
 
     // Check directory name is for deleted recordings
-    bool bDeleted = StringUtils::StartsWith(strDirectoryPath, "/deleted");
-    strDirectoryPath.erase(0, bDeleted ? 8 : 7);
+    bool bDeleted = StringUtils::StartsWith(strDirectoryPath, "/" PVR_RECORDING_DELETED_PATH);
+    strDirectoryPath.erase(0, bDeleted ? sizeof(PVR_RECORDING_DELETED_PATH) : sizeof(PVR_RECORDING_ACTIVE_PATH));
 
-    // get the directory structure if in non-flatten mode
-    if (m_bGroupItems)
-      GetSubDirectories(strDirectoryPath, &items, bDeleted);
+    // Get the directory structure if in non-flatten mode
+    // Deleted view is always flatten. So only for an active view
+    if (!bDeleted && m_bGroupItems)
+      GetSubDirectories(strDirectoryPath, &items);
```

Comment:
If we do not call GetSubDirectory() in case we are in the deleted view why do we need to differ between deleted/active items in GetSubDirectory()